boa/rpc.py

```diff
@@ -54,7 +55,24 @@ def from_json(cls, data):
         return cls(message=data["message"], code=data["code"])
 
 
-class EthereumRPC:
+class RPC:
+    """
+    Base interface for RPC implementations.
+    This abstract class does not use ABC for performance reasons.
+    """
+
+    @property
+    def name(self) -> str:
+        raise NotImplementedError
+
+    def fetch(self, method: str, params: Any) -> Any:
+        raise NotImplementedError
+
+    def fetch_multi(self, payloads: list[tuple[str, Any]]) -> list[Any]:
+        raise NotImplementedError
+
+
+class EthereumRPC(RPC):
```

Comment:
we can probably add a comment to `raw_fetch_multi` -- it's not really necessary anymore because the inheritance tree has changed.

Reply:
https://github.com/vyperlang/titanoboa/pull/124/commits/e3a4b192dbd7d520e99c910e50af05c9b95614f6